azure-devops/azext_devops/dev/pipelines/pipeline.py

```diff
@@ -123,8 +130,36 @@ def pipeline_run(id=None, branch=None, commit_id=None, name=None, open=False, va
         raise ValueError('Either the --id argument or the --name argument ' +
                          'must be supplied for this command.')
     client = get_build_client(organization)
+
     if id is None:
         id = get_definition_id_from_name(name, client, project, folder_path)
+
+    if parameters:
+        client = get_new_pipeline_client_v60(organization)
+
+        repositories = {"self": RepositoryResourceParameters(ref_name=branch, version=commit_id)}
+        resources = RunResourcesParameters(repositories=repositories)
+
+        template_parameters = ast.literal_eval(parameters)
+
+        param_variables = None
+        if variables:
+            param_variables = {}
+            for variable in variables:
+                separator_pos = variable.find('=')
+                if separator_pos >= 0:
+                    param_variables[variable[:separator_pos]] = {"value": variable[separator_pos + 1:]}
+                else:
+                    raise ValueError('The --variables argument should consist of space separated "name=value" pairs.')
```

Comment:
similar logic seems to exist below as well, can we refactor code to have the logic only in one place?